Docs/service descriptions

norling opened the pull request:
Adds markdown descriptions of all pipeline services (to be extended later with further information), as well as a trigger Action to update the documentation at neicnordic/neic-sda with the content of these files.

To make the action work, a secret needs to be added named `NEIC_SDA_ACCESS`, containing a github token with the
"Actions: Read and Write" permission on the neicnordic/neic-sda repository. Could someone with more permissions than me sort that out?

blankdots reviewed (approved):
there are a few inconsistencies like Central EGA, central EGA and CentralEGA, but lgtm

kjellp commented:
NEIC_SDA_ACCESS secret yesterday.

norling commented:
Good catch about the inconsistent referencing to Central EGA @blankdots! :) I rebased the PR on master before opening and fixed this at the same time.

jbygdell commented on cmd/backup/backup.md:
```diff
@@ -0,0 +1,40 @@
+# sda-pipeline: backup
+
+Merges data with the encryption header and moves it to backup storage.
+
```
Only if `copyHeader` is enabled

aaperis commented on cmd/pipeline.md:
```diff
@@ -0,0 +1,23 @@
+
+There are also three additional support services:
+
+1. [Backup](backup.md) copies data from archive storage to backup storage, reencrypting the header.
```
```suggestion
1. [Backup](backup.md) copies data from archive storage to backup storage, optionally re-encrypting and re-attaching the headers.
```

aaperis commented on cmd/pipeline.md:
```diff
@@ -0,0 +1,23 @@
+
+1. [Backup](backup.md) copies data from archive storage to backup storage, reencrypting the header.
+1. [Intercept](intercept.md) relays messages from Central EGA to the system.
+1. [Notify](notify.md) sends user e-mail messages.
```
```suggestion
1. [Notify](notify.md) sends e-mail messages to users.
```

aaperis commented on cmd/verify/verify.md:
```diff
@@ -0,0 +1,48 @@
+The verify service ensures that ingested files are encrypted with the correct key, and that the provided checksums match those of the ingested files.
+
+When running, verify reads messages from the configured RabbitMQ queue (default: "archived").
+For each message, these steps are taken (if not otherwise noted, errors halts progress and the service moves on to the next message.
```
```suggestion
For each message, these steps are taken (if not otherwise noted, errors halt progress and the service moves on to the next message.
```

aaperis commented on cmd/verify/verify.md:
```diff
@@ -0,0 +1,48 @@
+If the message can’t be validated it is discarded with an error message in the logs.
+
+1. The service attempts to fetch the header for the file id in the message from the database.
+If this fails a NACK will be sent for the RabbitMQ message, the error will be written to the logs, and send to the RabbitMQ error queue.
```
```suggestion
If this fails a NACK will be sent for the RabbitMQ message, the error will be written to the logs, and sent to the RabbitMQ error queue.
```

aaperis commented on cmd/verify/verify.md:
```diff
@@ -0,0 +1,48 @@
+1. The file size, md5 and sha256 checksum will be read from the decryptor.
+If this fails an error will be written to the logs.
+
+1. If the `re_verify` bool is not set in the RabbitMQ message, the message processing ends here, and continues with the next message.
```
```suggestion
1. If the `re_verify` boolean is not set in the RabbitMQ message, the message processing ends here, and continues with the next message.
```

aaperis reviewed (approved):
This is nicely descriptive! Few typos (some small others smaller) pilled up for fixing, but pre-approved :-).

jbygdell commented on cmd/backup/backup.md:
```diff
@@ -0,0 +1,40 @@
+Merges data with the encryption header and moves it to backup storage.
+
+## Service Description
+The backup service copies files from the archive storage, reattaches the encryption header, and writes the complete file to backup storage.
```
```suggestion
The backup service copies files from the archive storage to backup storage. If a public key is supplied the header will be re-encrypted and attached to the file before writing it to backup storage.
```
